### PR TITLE
Changes to the portal, build and deployment files for new log animation 

### PR DESCRIPTION
**Pull request by brucenguyen1**
This PR is to prepare the Portal infrastructure before adding the new log animation plugin. New implementation is added to the Portal as follows (refer to https://apromore.atlassian.net/wiki/spaces/AP/pages/270926208/Log+Animation+-+New)
- DataChannelServlet: a servlet to allow back and forth data communication between browser and plugins
- PluginExecution: represents one execution instance of portal plugin, identified by a unique pluginExecutionId.
- PluginExecutionManager: allows to manage all running plugin executions on the portal

Changes to web.xml for DataChannelServlet
Changes to BaseController and MainController to use plugin execution instances.
Changes to pom.xml, build.xml, apromore-core.plan the new API module: log-animation-portal-plugin-api

This PR doesn't depend on any other PRs, but the coming PR to add the new log animation depends on this PR.

### CODE REVIEW

File: Apromore-Core-Components/Apromore-Portal/src/main/webapp/WEB-INF/web.xml [hunk 1]
```diff
@@ -33,7 +33,6 @@
   <context-param>
     <param-name>contextClass</param-name>
     <param-value>org.eclipse.virgo.web.dm.ServerOsgiBundleXmlWebApplicationContext</param-value>
-    <!-- <param-value>org.springframework.osgi.web.context.support.OsgiBundleXmlWebApplicationContext</param-value> -->
```
Comment (raboczi) [inline Apromore-Core-Components/Apromore-Portal/src/main/webapp/WEB-INF/web.xml:36]: This commented-out code was left in there deliberately.  Making the Virgo context class continue to work under Karaf involves a hack I'd like to remove ASAP.  The commented-out version is the non-Virgo-specific alternative that doesn't require the hack.

**Review by raboczi — APPROVED** [review]
While I'm okay with merging this, I'm a little weirded out that we now have an org.apromore.plugin.portal package in the portal-api component, and an org.apromore.portal.plugin package in the portal.

**Comment by brucenguyen1** [comment]
> While I'm okay with merging this, I'm a little weirded out that we now have an org.apromore.plugin.portal package in the portal-api component, and an org.apromore.portal.plugin package in the portal.

That's a valid point, "org.apromore.portal.plugin" may sound like it is the namespace for Portal Plugins (which is "org.apromore.plugin.portal"), but it is only for grouping classes used to manage the plugins, not to put plugins themselves. It's important to choose a good name in the first place.

Some suggested names instead of "org.apromore.portal.plugin":
- org.apromore.portal.plugin_management: underscore is accepted (https://docs.oracle.com/javase/specs/jls/se12/html/jls-6.html#jls-6.1), but it is uncommon.
- org.apromore.portal.pluginManagement: will create trouble with Unix and Windows.
- org.apromore.portal.pluginmanagement: looks unnecessarily long, and a bit hard to read and type fully.
- org.apromore.portal.pluginmgt: use abbreviation, but doesn't look good.
- org.apromore.portal.pluginman: not ideal, but seems ok.

I would go with "org.apromore.portal.pluginman".

**Comment by raboczi** [comment]
> I would go with "org.apromore.portal.pluginman".

How about org.apromore.portal.datachannel?

File: Apromore-Core-Components/Apromore-Portal/src/main/webapp/WEB-INF/web.xml (hunk 1 above)
Comment (brucenguyen1) [reply Apromore-Core-Components/Apromore-Portal/src/main/webapp/WEB-INF/web.xml:36]: Just don't get it, a commented-out code was left in there _deliberately_? It cannot run, it is not being used. Did you mean it is kept there as a note for you to use it in the future. If so, github keeps track of these changes, like a journal book. Like accountants, they don't keep notes on the side of their accounting book, but they use a general ledger.

**Comment by brucenguyen1** [comment]
> > I would go with "org.apromore.portal.pluginman".
> 
> How about org.apromore.portal.datachannel?

"org.apromore.portal.datachannel" is about the data channel servlet, but we're looking for a package name for PluginExecution and PluginExecutionManager classes.

File: Apromore-Core-Components/Apromore-Portal/src/main/webapp/WEB-INF/web.xml (hunk 1 above)
Comment (raboczi) [reply Apromore-Core-Components/Apromore-Portal/src/main/webapp/WEB-INF/web.xml:36]: This is a configuration file, and that is an alternative configuration.  Please don't remove it.  I can write a lengthy comment warning people not to remove it and why, if you prefer.

Comment (brucenguyen1) [reply Apromore-Core-Components/Apromore-Portal/src/main/webapp/WEB-INF/web.xml:36]: Ok, I'll add it back, and pls write some comment hint saying it's important to leave it there, for many people reading that.

**Comment by brucenguyen1** [comment]
> > > I would go with "org.apromore.portal.pluginman".
> > 
> > 
> > How about org.apromore.portal.datachannel?
> 
> "org.apromore.portal.datachannel" is about the data channel servlet, but we're looking for a package name for PluginExecution and PluginExecutionManager classes.

**Comment by brucenguyen1** [comment]
Accidentally clicked the close button.

**Comment by brucenguyen1** [comment]
> > > > I would go with "org.apromore.portal.pluginman".
> > > 
> > > 
> > > How about org.apromore.portal.datachannel?
> > 
> > 
> > "org.apromore.portal.datachannel" is about the data channel servlet, but we're looking for a package name for PluginExecution and PluginExecutionManager classes.

"org.apromore.portal.plugincontrol" would be short enough, easy to understand and avoid the confusion with the plugin API namespace.

**Pull request merged**